### ISSUE TITLE
Software HMAC-SHA-256 should use board SHA-256 when available

### ISSUE THREAD

**Issue by ia0** [google/wasefire#157]
When enabling software HMAC-SHA-256, the SHA-256 used as part of the HMAC is also done in software, even though there might be a non-software SHA-256 implementation in the board.

This is blocked by #156.

As part of fixing this issue, it would be worth distinguishing between `Unsupported` and `Software`. This permits getting a compile error when asking for `Software` without the correct dependencies. It shouldn't be a usability issue by providing macros to select the correct implementation based on a feature. See https://github.com/ia0/wasefire/tree/issue-157 for the initial attempt.

**Comment by ia0** [comment]
This was fixed by #174. I'm going to ignore splitting `Unsupported` and `Software`. A separate issue may be opened if this is believed to be useful enough. Right now convenience is more useful.
